File: backend/app/io/appointment_data.py
from sqlmodel import SQLModel, Field
from datetime import date
from typing import Optional

class Appointment_data(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    id_patient: str = Field(foreign_key="patient_data.identity_document")
    id_doctor: str = Field(foreign_key="doctor_data.credentials")
    title: str
    department: str
    attendance_date: date
    reason: str
    conclusion: Optional[str]

File: frontend/src/app/shared/components/appointment/appointmentData.ts
export interface AppointmentData {
    id: number;
    id_patient: string;
    id_doctor: string;
    title: string;
    doctorName: string;
    doctorIcon: string;
    patientName: string;
    patientIcon: string;
    description: string;
    department: string;
    date: Date;
    paid: boolean;
    active: boolean;
    event_id: string;
}